### DISCUSSION TITLE
phase issues from ploting EBSD map

### DISCUSSION THREAD

**Discussion by yuehengzhang95** [mtex-toolbox/mtex#1434] [Ask Anything]
Hi I have a question regarding to the ebsd map. 

I used the "import_wizard" function to import the .ang file. But when I plot the EBSD map, it seems to convert the crystals phases into the substrate phases and convert the substrate phases into the crystals phases. I would like to know why this happens and how to solve this.
![wrong_phase](https://user-images.githubusercontent.com/59850513/169711141-33e5071b-53c2-47b5-8164-eece6ce8acc1.jpg)

**Comment by kilir** [comment]
Hi,
So you say that gold should be green and the other phase blue?
Cheers,
Rüdiger

**Reply by yuehengzhang95** [reply]
Yes, I used the import_wizard function, then choose "EBSD", and then I set the parameters and color as I showed in the following figures. 
<img width="500" alt="Screen Shot 2022-06-14 at 2 25 30 PM" src="https://user-images.githubusercontent.com/59850513/173662112-7f05a8b3-0de5-4423-a7c7-931a2105720a.png">
<img width="497" alt="Screen Shot 2022-06-14 at 2 26 21 PM" src="https://user-images.githubusercontent.com/59850513/173662265-77d7e548-03b3-4bde-bb27-27ae1ea1fbee.png">
<img width="495" alt="Screen Shot 2022-06-14 at 2 27 18 PM" src="https://user-images.githubusercontent.com/59850513/173662473-24cea5bc-aa80-4e09-9d6f-1c5ecb3f6648.png">


Then, it automatically generate the codes for me, I have attached the codes here and then I plot the EBSD map. Can you check this for me, because I do not know why the phases are wrong ? Thanks!

`%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('432', [4.1 4.1 4.1], 'mineral', 'Gold', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('432', [3.9 3.9 3.9], 'mineral', 'Strontium Titanium Oxide', 'color', [0.56 0.74 0.56])};

% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = '/Users/yuehengzhang/Desktop/CMU/LauePUP422/5-11-22';

% which files to be imported
fname = [pname '/Longscan1.ang'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertSpatial2EulerReferenceFrame');


%%


ebsd = ebsd(ebsd.ci>0.01);


% reconstruct grains
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'unitcell');
ebsd(grains(grains.grainSize<=3)) = [];
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',5*degree);

% chose the correct crystal shape (cubic, hex are generic forms)
cS = ebsd('Gold').CS;

% select only large grains
largeGrains = grains(grains.grainSize>10)

% and plot the crystal shapes
% hold on
plot(largeGrains,cS)
% hold off
`

**Reply by yuehengzhang95** [reply]
This is what the generated "EBSD" map looks like.
<img width="389" alt="Screen Shot 2022-06-14 at 2 33 21 PM" src="https://user-images.githubusercontent.com/59850513/173663419-1950934d-9fbd-4419-a183-09b6791d45a5.png">

**Reply by kilir** [reply]
I do not really see what should be going wrong. Can you elaborate?

**Reply by yuehengzhang95** [reply]
The color representations are weird. I think gold should be BLUE and the other phase Green. This is what I set in "import_wizard" function as described in the first two screenshots.

**Comment by yuehengzhang95** [comment]
Yes!

**Comment by macosta541** [comment]
I have this problem when I am using an mlx file where I copy and pasted the output script from the import wizard for one file and then just changed the file name in the mlx file instead of re-running the import script. Basically, when I:

1) run import wizard for file A
2) copy and paste the output script into an mlx file 
3) write code as normal in mlx file
4) decide I want to run file B through the mlx file but only change the file name from the output script and do not re-run the import wizard separately for file B and then copy and paste
5) realize all my phases are wrong

the solution is to re-run the import wizard for each file and copy that into the mlx.  It's annoying but trivial.  I imagine it has to do with the order of phases in the cpr files and how the import wizard brings them in.

**Reply by kilir** [reply]
Are phases picked up correctly using  `ebsd = EBSD.load(yourfile) ` ?

**Reply by macosta541** [reply]
not when I use the import wizard output in an .mlx file  .   

I think it's the order that the CS list is in that's dictating the way it's reading the phases in.  

For example, for this .cpr file the order that the phases are put in is "forsterite - calcite - dolomite":
`StructureName=Calcite
Reference=[Calcite.cry]
Enabled=True
a=4.9900
b=4.9900
c=17.0640
alpha=90.00
beta=90.00
gamma=120.00
LaueGroup=7
ID1=
ID2=
[Phase3]
StructureName=Dolomite
Reference=[Dolomite.cry]
Enabled=True
a=4.8069
b=4.8069
c=16.0034
alpha=90.00
beta=90.00
gamma=120.00
LaueGroup=6`
...
So when I run the import wizard the output looks like this: 

`% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('mmm', [4.8 10 6], 'mineral', 'Forsterite', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('-3m1', [5 5 17], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Calcite', 'color', [0.56 0.74 0.56]),...
  crystalSymmetry('-3', [4.8 4.8 16], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Dolomite', 'color', [0.85 0.65 0.13])};


% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\filePath';

% which files to be imported
fname = [pname '\Map4_94JUB39.cpr']


%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','crc',...
  'convertEuler2SpatialReferenceFrame');
`

and if I change the .cpr file to one where the phases were added in a different order, such as "calcite-dolomite-forsterite" without changing the order that the CS file was read in, it will still pick up the first phase (in this case it's really calcite) as forsterite, the second phase (in this case dolomite) as calcite, and the third phase as dolomite (but it's really the forsterite)


For visual comparison, here is what it should be: 

![image](https://user-images.githubusercontent.com/74112001/170651260-dd9f6293-2e10-451c-a02e-9752916c565f.png)

and here is what it is when it is confused:

![image](https://user-images.githubusercontent.com/74112001/170651413-d22f812b-695e-4b04-be37-a3b1a3867f35.png)

**Reply by kilir** [reply]
This is expexted behavior. There's actually nothing wrong, the input data was just not matching the input script you used.
May I ask why you prefer the script you use over `EBSD.load()` ?
Cheers,
Rüdiger

**Comment by lfgmorales** [comment]
Still it is unclear what the problem is...


in your phase map, you have a blue phase that is labelled as "gold" (the dominant color) and a green phase which is labelled "strontium titanium oxide" (that you have about 20 grains) - this is what you setup and this is the resulting plot you sent before...




Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
HPM C57.3

Otto-Stern-Weg 3

8093 Zürich

Switzerland

Phone: + 41 44 633 3746

<tel:+41446333746>E-mail: ***@***.***



"No trees were destroyed in the sending of this message, however, a significant number of electrons were terribly inconvenienced"


________________________________

**Reply by yuehengzhang95** [reply]
The dominant phases are substrate which should be "strontium titanium oxide", and the crystals are actually gold which should be 20 grains here. This is what we observed in SEM, so I do not where I set wrong, sorry for the confusion...
![](https://user-images.githubusercontent.com/59850513/173663419-1950934d-9fbd-4419-a183-09b6791d45a5.png)

**Reply by kilir** [reply]
Can you provide a small example data set?

**Comment by yuehengzhang95** [comment]
Hi, this is the raw data. Thanks!
 Longscan1.ang
<https://drive.google.com/file/d/1DsFCcj0rPMTLfTPX-K0pZiGymgunr3Cu/view?usp=drive_web>


Rüdiger Kilian ***@***.***> 于2022年6月15日周三 15:03写道：

**Reply by kilir** [reply]
There's everything right with your file and importing. In the ang, Gold is the first phase and has phase 1,  Tausonite is the second phase and is phase 2. The same way they are imported into mtex.

Unless there is a magic switch somewhere stated in the file (from which I've never heard) that assignes the first phase the id 2 and second phase id 1, there's nothing wrong with it. So the file reports the "background" as quite heterogenously indexed background and blobs of SrTiO3.


![newfile](https://user-images.githubusercontent.com/11697557/173924884-90e23202-a090-4dd1-91a6-0ebf80a8283e.jpg)

**Reply by yuehengzhang95** [reply]
I am still a little confused about this. We should get blobs of Gold crystals, and the "background" should be SrTiO3. However, the blobs shown here are SrTiO3 while the background is Gold. I do not understand why this happens? Can you ellaborate this for me, many thanks!

**Comment by zmichels** [comment]
Hi, The data are what the data are… and as Rüdiger indicated, the raw data comprise a heterogeneously indexed background/matrix with larger homogenous regions indexed as SrTiO3. If it seems to you those data do not represent your samples microstructure, then something may have went wrong during data acquisition (which has nothing to do with MTEX). Besides EBSD, have you done any other characterization on this sample – like some other imaging/observation that set your expectation (for the opposite result)? Without knowing your scenario, I first image a couple possibilities: [1] Something was off with the acquisition system (maybe you could try opening the data with that software and plot the phase map to see), [2] The data accurately represent the sample region you mapped, and it just isn’t what you thought it would be.  Zach    From: yuehengzhang95Sent: Wednesday, June 15, 2022 2:15 PMTo: mtex-toolbox/mtexCc: SubscribedSubject: Re: [mtex-toolbox/mtex] phase issues from ploting EBSD map (Discussion #1434) I am still a little confused about this. We should get blobs of Gold crystals, and the "background" should be SrTiO3. However, the blobs shown here are SrTiO3 while the background is Gold. I do not understand why this happens? Can you ellaborate this for me, many thanks!—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

**Comment by yuehengzhang95** [comment]
Hi, I just used TSL/OIM software to look at the dataset, it occurs to me
that the phase map matches with my expectation because the larger
homogenous regions (large green blobs) indexed as gold instead of SrTiO3!
Thus, I do not think the data acquisition is wrong.  Also, I attached my
SEM image here just for reference. So, is there anything I need to do to
correct the phases in MTEX? Many thanks!

[image: image.png]
[image: image.png]

Best,
Yueheng



Zachary Michels ***@***.***> 于2022年6月15日周三 17:30写道：

**Reply by kilir** [reply]
I think something went wrong with that image attachment.

**Reply by yuehengzhang95** [reply]
Let me try to upload again!
![image-2](https://user-images.githubusercontent.com/59850513/174337169-1e69b517-64be-4bd6-b0a1-404d272e6ad4.png)
![image-3](https://user-images.githubusercontent.com/59850513/174337142-4d40e4d9-8639-43bb-969e-60cb97b6673b.png)

**Reply by zmichels** [reply]
What does an image of the pixels look like from MTEX before any processing (as opposed to the grain reconstruction image you sent previously)?

**Comment by zmichels** [comment]
If you plot the pixels in MTEX (rather than your grain reconstruction), what does the map look like?

Zach

**Comment by zmichels** [comment]
Is there a way to reopen your data on the acquisition system/software? …to check? …and maybe retry the file export? It looks like MTEX is doing exactly what you ask, so I keep wondering if there is something about the initial dataset that is leading to this result(?).

Zach
